ThunkLibs/libxcb/libxcb_Guest.cpp

```diff
@@ -30,6 +30,8 @@ tags: thunklibs|xcb
 
 static std::thread CBThread{};
 static std::atomic<bool> CBDone{false};
+static std::mutex CBThreadMutex;
```

Comment:
Are these changes related to the other changes in this PR? If not, would be better suited in a dedicated PR since it's already a fairly large bunch of changes as-is.

Reply:
It's related to this PR in that this PR completely breaks xcb thunking if this isn't in place.
We could move it out to a PR that needs to be merged before this one?

Reply:
Different PR sounds good, just to make it clear this one really just handles `std->fextl` migration itself. Nothing egregious about the change itself so it should be an easy merge.

Reply:
#2583 opened and that commit dropped from this PR. That PR needs to be merged first.